cf/data/creation.py

```diff
@@ -3,81 +3,47 @@
 from dask.base import is_dask_collection
-from dask.utils import SerializableLock
 
 
-def convert_to_builtin_type(x):
-    """Convert a non-JSON-encodable object to a JSON-encodable built-in
-    type.
-
-    Possible conversions are:
-
-    ================  =======  ================================
-    Input             Output   `numpy` data-types covered
-    ================  =======  ================================
-    `numpy.bool_`     `bool`   bool
-    `numpy.integer`   `int`    int, int8, int16, int32, int64,
-                               uint8, uint16, uint32, uint64
-    `numpy.floating`  `float`  float, float16, float32, float64
-    ================  =======  ================================
-
-    .. versionadded:: 4.0.0
-
-    :Parameters:
-
-        x:
-            TODO
-
-    :Returns:
-
-            TODO
-
-    **Examples**
-
-    >>> type(_convert_to_netCDF_datatype(numpy.bool_(True)))
-    bool
-    >>> type(_convert_to_netCDF_datatype(numpy.array([1.0])[0]))
-    double
-    >>> type(_convert_to_netCDF_datatype(numpy.array([2])[0]))
-    int
-
-    """
-    if isinstance(x, np.bool_):
-        return bool(x)
-
-    if isinstance(x, np.integer):
-        return int(x)
-
-    if isinstance(x, np.floating):
-        return float(x)
-
-    raise TypeError(f"{type(x)!r} object is not JSON serializable: {x!r}")
-
-
-def to_dask(array, chunks, default_chunks=False, **from_array_options):
-    """TODODASKDOCS.
+def to_dask(array, chunks, **from_array_options):
+    """Create a `dask` array.
 
     .. versionadded:: TODODASKVER
 
     :Parameters:
 
         array: array_like
-            TODODASKDOCS.
+            The array to be converted to a `dask` array. Examples of
+            valid types include `numpy` arrays, `dask` arrays, `Array`
+            subclasses, `list`, `tuple`, scalars.
 
         chunks: `int`, `tuple`, `dict` or `str`, optional
-            Specify the chunking of the returned dask array.
-
-            Any value accepted by the *chunks* parameter of the
+            Specify the chunking of the returned dask array.  Any
+            value accepted by the *chunks* parameter of the
             `dask.array.from_array` function is allowed.
 
-        dask_from_array_options: `dict`
+            Ignored if *array* is a `dask` array, which already
+            defines its own chunks.
+
+            Might get automatically modified if *array* is a
+            compressed `Array` subclass.
+
+        from_array_options: `dict`, optional
             Keyword arguments to be passed to `dask.array.from_array`.
 
+            If *from_array_options* has no ``'lock'`` key then the
+            `lock` keyword is set to the `_dask_lock` attribute of
+            *array* or, if there is no such attribute, `False`.
+
+            If *from_array_options* has no ``'meta'`` key then the
+            `meta` keyword is set to the `_dask_meta` attribute of
+            *array* or, if there is no such attribute, `None`.
```

Comment:
Is there a particular reason we have the former becoming `False` whereas the latter becoming `None` "if there is no such attribute" - I guess it stems from the [different defaults in input and motivation for the `meta` and `lock` keywords to `da.from_array`](https://docs.dask.org/en/stable/generated/dask.array.from_array.html#dask-array-from-array)), in which case maybe we can briefly state that to be clear that this is the reason we default to those too? Something like:

```suggestion
            If *from_array_options* has no ``'lock'`` key then the
            `lock` keyword is set to the `_dask_lock` attribute of
            *array* or, if there is no such attribute, `False`,
            the value it is assigned to by default in
            `dask.array.from_array`.

            If *from_array_options* has no ``'meta'`` key then the
            `meta` keyword is set to the `_dask_meta` attribute of
            *array* or, if there is no such attribute, `None`,
            the value it is assigned to by default in
            `dask.array.from_array`.
```

Reply:
In hindsight, maybe this is over the top. Let me know what you think...

Reply:
Fair enough, marking as resolved!